### DISCUSSION TITLE
compileCommands overwrite defines in c_cpp_properties.json?

### DISCUSSION THREAD

**Discussion by mkitby** [microsoft/vscode-cpptools#9390] [Q&A]
Hi,

There are some internal preprocessors in cross-compiling compiler, and these preprocessors are not present in compile_commands.json. I defined these in defines field in c_cpp_properties.json. But it cannot work. Then I found this, 

> If compileCommands has a valid value and the file open in the editor is in the database, use the compile command in the database entry to determine the include path and defines.

why overwrite defines? how to avoid this?

**Comment by sean-mcmanus** [comment]
You can merge your additional defines with the ones fro the compileCommands provider by setting the "mergeConfigurations" property to true. See the implementation at https://github.com/microsoft/vscode-cpptools/pull/8174 .

**Reply by mkitby** [reply]
It still does not work when configure "C_Cpp.default.mergeConfigurations": true,  __TI_ARM__ is still not defined


```
{
    "configurations": [
        {
            "name": "TI-CGT-ARM",
            "includePath": ["${workspaceFolder}/**"],
            "defines": ["__TI_ARM__"],
            "compilerPath": "",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "gcc-arm",
            "compileCommands": "${workspaceFolder}/Output/compile_commands.json"
        }
    ],
    "version": 4
}
```

**Reply by sean-mcmanus** [reply]
Oops, sorry -- the "mergeConfigurations" is only supported with a configurationProvider and support for compileCommands is being tracked by https://github.com/microsoft/vscode-cpptools/issues/6909 .
